lib/matplotlib/_constrained_layout.py

```diff
@@ -175,12 +175,20 @@ def do_constrained_layout(fig, renderer, h_pad, w_pad,
                 _make_layout_margins(ax, renderer, h_pad, w_pad)
 
         # do layout for suptitle.
-        if fig._suptitle is not None and fig._suptitle._layoutbox is not None:
-            sup = fig._suptitle
-            bbox = invTransFig(sup.get_window_extent(renderer=renderer))
+        suptitle = fig._suptitle
+        do_suptitle = (suptitle is not None and
+                       suptitle._layoutbox is not None and
+                       suptitle.get_in_layout())
+        if do_suptitle:
+            suptitle = fig._suptitle
+            bbox = invTransFig(
+                suptitle.get_window_extent(renderer=renderer))
             height = bbox.y1 - bbox.y0
             if np.isfinite(height):
-                sup._layoutbox.edit_height(height+h_pad)
+                # reserve at top of figure include an h_pad above and below
+                suptitle._layoutbox.edit_height(height + h_pad * 2)
+        else:
+            do_suptitle = False
```

Comment:
This line seems odd - won't it only be triggered if `do_suptitle == False`, and is therefore redundant?

Reply:
Yep, sorry!